#! /bin/sh /usr/share/dpatch/dpatch-run
## 015-empty-passwords.dpatch by David Prévot <david@tilapin.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Adjust behaviour of Auth_MySQL_Empty_Passwords

diff -urNad mod-auth-mysql~/mod_auth_mysql.c mod-auth-mysql/mod_auth_mysql.c
--- mod-auth-mysql~/mod_auth_mysql.c	2010-01-23 17:54:17.000000000 +0100
+++ mod-auth-mysql/mod_auth_mysql.c	2010-01-23 17:54:23.000000000 +0100
@@ -1504,10 +1504,16 @@
 	encryption_type_entry *ete;
 	
 	/* empty password support */
-	if (sec->allow_empty_passwords && !strlen(hashed)) {
-		APACHELOG(APLOG_INFO, r, "User successful on empty password");
-		return 1;
-	}
+	if (!strlen(hashed)) {
+               if (sec->allow_empty_passwords) {
+                       APACHELOG(APLOG_INFO, r, "User successful on empty password");
+                       return 1;
+               } else {
+                       APACHELOG(APLOG_INFO, r, "Rejecting login because of empty password field in DB");
+                       return 0;
+               }
+        }
+
 			
 	for (ete=supported_encryption_types; ete->name; ete++) {
 		if (sec->encryption_types & ete->flag) {
